src/prefect/locking/protocol.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional, Protocol
+
+from typing_extensions import runtime_checkable
```

Comment:
I think you can safely import this from `typing` as it's been in `typing` since Python 3.8